Makefile.pre.in

```diff
@@ -2643,7 +2643,18 @@ config.status:	$(srcdir)/configure
 Python/asm_trampoline.o: $(srcdir)/Python/asm_trampoline.S
 	$(CC) -c $(PY_CORE_CFLAGS) -o $@ $<
 
-Python/jit.o: regen-jit
+
+JIT_DEPS = \
+		$(srcdir)/Python/jit.c \
+		$(srcdir)/Python/executor_cases.c.h \
+		pyconfig.h \
+		$(srcdir)/Tools/jit/template.c \
```

Comment:
`jit_stencils.h` doesn't depend on `jit.c` (it's the other way around.

If there's no way to express a dependency on *everything* inside of `Tools/jit` (recursive), maybe we can just approximate it two levels deep with something like this?

```suggestion
JIT_DEPS = \
		$(srcdir)/Tools/jit/*/* \
		$(srcdir)/Tools/jit/* \
		$(srcdir)/Python/executor_cases.c.h \
		pyconfig.h
```

Reply:
GNU Make 4.3 (from Ubuntu 22.04) doesn't like `$(srcdir)/Tools/jit/*/*` because there are no directories in `Tools/jit` right now, I'm not certain that you can use multiple globs in a Makefile path name, and that's a bit more liberal than I think we ought to be here anyway.

Point taken about `jit.c`, though; removed it from `JIT_DEPS`, added `$(srcdir)/Tools/jit/*.py`, and switched from `template.c` to `*.c`.  If and when things get rearranged into a hierarchy, the rules should be fairly straightforward to update.